packages/expo-asset/plugin/src/withAssetsIos.ts

```diff
@@ -0,0 +1,134 @@
+import { ExpoConfig } from '@expo/config-types';
+import { generateImageAsync } from '@expo/image-utils';
+import {
+  ContentsJsonImage,
+  createContentsJsonItem,
+  writeContentsJsonAsync,
+} from '@expo/prebuild-config/build/plugins/icons/AssetContents';
+import {
+  ConfigPlugin,
+  IOSConfig,
+  InfoPlist,
+  XcodeProject,
+  withInfoPlist,
+  withXcodeProject,
+} from 'expo/config-plugins';
+import { ensureDir, writeFile } from 'fs-extra';
+import path from 'path';
+
+import { fontTypes, imageTypes, resolveAssetPaths, validateAssets } from './utils';
+
+const IMAGE_DIR = 'Images.xcassets';
+
+export const withAssetsIos: ConfigPlugin<string[]> = (config, assets) => {
+  config = addAssetsToTarget(config, assets);
+  config = addFontsToPlist(config, assets);
+  return config;
+};
+
+function addAssetsToTarget(config: ExpoConfig, assets: string[]) {
+  return withXcodeProject(config, async (config) => {
+    const resolvedAssets = await resolveAssetPaths(assets, config.modRequest.projectRoot);
+    const validAssets = validateAssets(resolvedAssets);
+    const project = config.modResults;
+    const platformProjectRoot = config.modRequest.platformProjectRoot;
+    IOSConfig.XcodeUtils.ensureGroupRecursively(project, 'Resources');
+
+    const images = validAssets.filter((asset) => imageTypes.includes(path.extname(asset)));
+    const assetsForResourcesDir = validAssets.filter(
+      (asset) => !imageTypes.includes(path.extname(asset))
+    );
+
+    await addImageAssets(images, config.modRequest.projectRoot);
+    addResourceFiles(project, platformProjectRoot, assetsForResourcesDir);
+
+    return config;
+  });
+}
+
+function addResourceFiles(project: XcodeProject, platformRoot: string, assets: string[]) {
+  for (const asset of assets) {
+    const assetPath = path.relative(platformRoot, asset);
+    IOSConfig.XcodeUtils.addResourceFileToGroup({
+      filepath: assetPath,
+      groupName: 'Resources',
+      project,
+      isBuildFile: true,
+      verbose: true,
+    });
+  }
+}
+
+async function addImageAssets(assets: string[], root: string) {
+  const iosNamedProjectRoot = IOSConfig.Paths.getSourceRoot(root);
+  for (const asset of assets) {
+    const name = path.basename(asset, path.extname(asset));
+    const image = path.basename(asset);
+
+    const assetPath = path.resolve(iosNamedProjectRoot, `${IMAGE_DIR}/${name}.imageset`);
+    await ensureDir(assetPath);
+
+    const buffer = await generateImageAsync({ projectRoot: root }, { src: asset } as any);
```

Comment:
why is this `any` cast needed? since the callsite is first-party, can we update the types to allow this call?

Reply:
We use this in various places in the plugins, splashscreens, icons etc. In this case, we don't need all the extra options and we use it like this in other places https://github.com/expo/expo/blob/8cc11815565476c5569359e8cbb7001ebec0d870/packages/%40expo/prebuild-config/src/plugins/unversioned/expo-splash-screen/withIosSplashAssets.ts#L151-L153

Probably would be best to expose an api to write to XCAssets. I didn't want to change the types and affect the implementation so just I stuck to what we have done before

Reply:
The issue with an `any` cast is it makes refactoring more dangerous. If someone were to want to refactor the argument types of `generateImageAsync` in the future it's likely that they'll miss this callsite since `tsc` won't catch it and we don't have tests for it, so it will cause a bug for our users that may go uncaught in an SDK.

I think the fact that there's more than one place that does this is even more reason to fix the types. If it were just one place it could be fine to just have a test, but because it's multiple places we should definitely fix it.

Reply:
Cool, I agree, I can speak to Cedric or Evan and follow up in a separate PR 👍

Reply:
Instead of `as any` you could do `as unknown as ImageOptions`, maybe?